core/templates/dev/head/services/PlaythroughIssuesService.ts

```diff
@@ -145,121 +145,96 @@ angular.module('oppia').factory('PlaythroughIssuesService', [
             controller: [
               '$scope', '$uibModalInstance', 'playthroughIndex',
               'playthrough', 'AlertsService', 'LearnerActionRenderService',
+              'ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS',
               function(
                   $scope, $uibModalInstance, playthroughIndex,
-                  playthrough, AlertsService, LearnerActionRenderService) {
+                  playthrough, AlertsService, LearnerActionRenderService,
+                  ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS) {
                 $scope.playthroughIndex = playthroughIndex;
 
-                $scope.displayBlocks =
-                  LearnerActionRenderService.getDisplayBlocks(
+                $scope.startingIndices =
+                  LearnerActionRenderService.getStartingIndices(
                     playthrough.actions);
-                $scope.reversedDisplayBlocks =
-                  $scope.displayBlocks.slice().reverse();
+                $scope.currentChoice = 0;
 
-                var blockActionIndexMapping = {};
-                $scope.displayBlocks.reduce(
-                  function(runningTotal, displayBlock, i) {
-                    blockActionIndexMapping[i] =
-                      runningTotal - displayBlock.length;
-                    return blockActionIndexMapping[i];
-                  }, playthrough.actions.length + 1);
+                $scope.isIssueMIS = false;
+                if (playthrough.issueType === (
+                  ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS)) {
+                  $scope.isIssueMIS = true;
+                  var lars = LearnerActionRenderService;
+                  $scope.issueMisTableHtml =
+                    lars.renderFinalDisplayBlockForMISIssueHTML(
+                      playthrough.actions.slice($scope.startingIndices[0]),
+                      $scope.startingIndices[0] + 1);
+                }
 
-                $scope.maxHidden = $scope.displayBlocks.length - 1;
+                $scope.maxHidden = $scope.startingIndices.length - 1;
 
-                $scope.getDisplayBlockIndex = function(displayBlock) {
-                  return $scope.displayBlocks.indexOf(displayBlock);
-                };
-
-                $scope.isDisplayBlockOnInitDisplay = function(block) {
-                  return $scope.getDisplayBlockIndex(block) === 0;
-                };
-
-                $scope.createDisplayBlockNavId = function(block) {
-                  return $scope.getDisplayBlockIndex(block) + 1;
+                /**
+                 * Returns the list of learner actions to currently be
+                 * displayed in the playthrough modal.
+                 * @returns {LearnerAction[]}
+                 */
+                $scope.getActionsToRender = function() {
+                  if ($scope.isIssueMIS) {
```

Comment:
@brianrodri For Multiple Incorrect Submission issues, we display a table of the incorrectly submitted answers. So, in this case, the highlighted set of actions are processed separately.

Reply:
Let's try to have the directive handle this (instead of manually building the HTML from strings). Or we could make a new mini-directive for this table.